### PR TITLE
📚 Feature/readme

### PR DESCRIPTION
**Pull request by ShwetaChauhan18**
- 📚 Feature/readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,167 @@
...
+| common |
+|| child | Widget | Replace PlaceHolder with your widget | none |
+|| showPlaceHolder | bool | PlaceHolder visibility | none |
+|| titleTextStyle | TextStyle | Title textstyle | TextStyle(fontSize: 20, letterSpacin.g: 0.4, fontWeight: FontWeight.bold) |
+|| subTitleTextStyle | TextStyle | Subtitle textstyle | TextStyle(fontSize: 14, letterSpacing: 0.4) |
+|| onButtonClick | VoidCallback | On click of button | none ||
+
+## Other Library used:
+
+* __[spinkit](https://pub.dev/packages/flutter_spinkit)__
```
Comment (vatsaltanna) [inline README.md:140]: few other libraries are missing here

File: README.md [hunk 2]
```diff
@@ -1,16 +1,170 @@
...
+2. For loader change :
+    - Use SpinKit loader name: e.g `LoaderName.hourGlass`
+    - LoaderName is Enum in which all SpinKit Loader name added.
+
+## How to use
+
+Check out the **example** app in the [example](example) directory or the 'Example' tab on
+pub.dartlang.org for a more complete example.
+
+Methods defined by `SSPlaceholder` class:
```
Comment (AnkitSimform) [inline README.md:99]: Can we merge the row for the section title?

File: README.md [hunk 3]
```diff
@@ -1,16 +1,170 @@
...
+[![Build](https://github.com/SimformSolutionsPvtLtd/flutter_showcaseview/workflows/Build/badge.svg?branch=master)](https://github.com/SimformSolutionsPvtLtd/flutter_showcaseview/actions) [![ssplaceholderview](https://img.shields.io/pub/v/showcaseview?label=showcaseview)](https://pub.dev/packages/showcaseview)
 
-## Getting Started
+A Flutter package allows you to add placeholder content while loading or empty items.
 
-This project is a starting point for a Flutter application.
+## Preview
 
-A few resources to get you started if this is your first Flutter project:
+<img src="https://github.com/SimformSolutionsPvtLtd/flutter_ss_placeholder_view/blob/feature/readme/preview/placeholderview.gif" width="400" height="711">
```
Comment (HappyMakadiyaS) [inline README.md]: Provide master branch url instead of feature/readme branch.

File: README.md (hunk 2 above)
Comment (HappyMakadiyaS) [reply README.md:99]: ^^
Ref: https://stackoverflow.com/a/61997043/19938097

File: example/lib/mail_page.dart [hunk 4]
```diff
@@ -64,6 +66,10 @@ class _MailPageState extends State<MailPage> {
     );
   }
 
+  void _clearData() {
```
Comment (HappyMakadiyaS) [inline example/lib/mail_page.dart]: convert to expression body

File: example/lib/mail_page.dart [hunk 5]
```diff
@@ -253,16 +242,21 @@ class _MailPageState extends State<MailPage> {
 
   void _changePlaceHolderImage(String? newValue) {
     if (newValue == 'Image') {
-      placeholderImage = 'assets/loading.jpeg';
+      placeholderWidget = null;
```
Comment (HappyMakadiyaS) [inline example/lib/mail_page.dart]: Can be convert into switch case

File: example/lib/main.dart [hunk 6]
```diff
@@ -25,13 +19,11 @@ class MyApp extends StatelessWidget {
...
             ),
+        scaffoldBackgroundColor: Colors.white,
       ),
       debugShowCheckedModeBanner: false,
       home: Scaffold(
-        body: SafeArea(
-          bottom: false,
-          child: Builder(builder: (context) => const MailPage()),
-        ),
+        body: Builder(builder: (context) => const MailPage()),
```
Comment (HappyMakadiyaS) [inline example/lib/main.dart]: Builder can be removed.

File: example/lib/mail_page.dart [hunk 7]
```diff
@@ -206,17 +194,18 @@ class _MailPageState extends State<MailPage> {
                 placeHolderImageConfig: PlaceHolderImageConfig(
                   extraPadding: const EdgeInsets.only(bottom: 10),
                   placeholderImage: placeholderImage,
+                  placeholderWidget: placeholderWidget,
```
Comment (HappyMakadiyaS) [inline example/lib/mail_page.dart]: Can't we remove the placeholder keyword from the parameter name? as class name itself indicating that its configuration of Placeholder

File: example/lib/mail_page.dart [hunk 8]
```diff
@@ -206,17 +194,18 @@ class _MailPageState extends State<MailPage> {
...
                   animationName: 'walk',
                 ),
                 loadingConfig: LoadingConfig(
-                  loadingTitle: 'Loading Title',
-                  loadingSubtitle: 'Loading subtitle',
+                  loadingTitle: 'Please Wait...',
+                  loadingSubtitle: 'Loading your data...',
```
Comment (HappyMakadiyaS) [inline example/lib/mail_page.dart]: ^^

File: lib/src/vector_asset.dart [hunk 9]
```diff
@@ -32,70 +32,41 @@ class VectorAsset extends StatelessWidget {
...
-                              )
-        : Image.asset('');
+                    : Image.asset(
+                        image!,
+                        width: width,
+                        height: height,
+                        fit: fit ?? BoxFit.contain,
+                        alignment: alignment,
+                      )
+            : Image.asset('');
```
Comment (HappyMakadiyaS) [inline lib/src/vector_asset.dart]: Can we use Sizedbox.shrink()?

**Review by HappyMakadiyaS — CHANGES_REQUESTED** [review]

**Comment by HappyMakadiyaS** [comment]
Also suggesting you to update PR title and branch name.

**Review by HappyMakadiyaS — COMMENTED** [review]
LGTM

**Pull request closed without merging**